--- libtheora-1.1.1/examples/encoder_example.c.orig	2015-01-19 20:00:34.333600000 +0300
+++ libtheora-1.1.1/examples/encoder_example.c	2015-01-19 20:02:00.570400000 +0300
@@ -47,11 +47,12 @@
 #include "vorbis/codec.h"
 #include "vorbis/vorbisenc.h"
 
-#ifdef _WIN32
+#if defined(_WIN32)
 /*supply missing headers and functions to Win32. going to hell, I know*/
 #include <fcntl.h>
 #include <io.h>
 
+#if !defined(__MINGW64_VERSION_MAJOR)
 static double rint(double x)
 {
   if (x < 0.0)
@@ -60,6 +61,7 @@
     return (double)(int)(x + 0.5);
 }
 #endif
+#endif
 
 const char *optstring = "b:e:o:a:A:v:V:s:S:f:F:ck:d:z:\1\2\3\4";
 struct option options [] = {
--- libtheora-1.1.1/configure.orig      2017-09-13 09:26:55.875971800 +0200
+++ libtheora-1.1.1/configure   2017-09-13 09:27:36.175729800 +0200
@@ -12059,7 +12059,7 @@
     THEORA_LDFLAGS="$THEORA_LDFLAGS -no-undefined"
     THDEC_VERSION_ARG="-export-symbols \$(top_srcdir)/win32/xmingw32/libtheoradec-all.def"
     THENC_VERSION_ARG="-export-symbols \$(top_srcdir)/win32/xmingw32/libtheoraenc-all.def"
-    THENC_VERSION_ARG="$THENC_VERSION_ARG -ltheoradec"
+    THENC_VERSION_ARG="$THENC_VERSION_ARG -ltheoradec"
     THC_VERSION_ARG="-export-symbols \$(top_srcdir)/win32/libtheora.def"
     { $as_echo "$as_me:$LINENO: result: -export-symbols" >&5
 $as_echo "-export-symbols" >&6; }
